src/store/participantsStore.js

```diff
@@ -114,6 +125,32 @@ const mutations = {
 			console.error(`The conversation you are trying to purge doesn't exist`)
 		}
 	},
+	setInCall(state, { token, sessionId, flags }) {
+		if (flags === PARTICIPANT.CALL_FLAG.DISCONNECTED) {
+			if (state.inCall[token] && state.inCall[token][sessionId]) {
+				Vue.delete(state.inCall[token], sessionId)
+			}
+
+			if (state.connecting[token] && state.connecting[token][sessionId]) {
+				Vue.delete(state.connecting[token], sessionId)
+			}
+		} else {
+			if (!state.inCall[token]) {
+				Vue.set(state.inCall, token, {})
+			}
+			Vue.set(state.inCall[token], sessionId, flags)
+
+			if (!state.connecting[token]) {
+				Vue.set(state.connecting, token, {})
+			}
+			Vue.set(state.connecting[token], sessionId, flags)
```

Comment:
so we're setting the exact same values/structure to "connecting" like "inCall" ?
either redundant or a copy-paste mistake ?

since none of these have any other getters than `isInCall` and `isConnected` the data structure could be merged into a single one but still keep both getters ?

Reply:
I find it easier with this explicit way instead of wrapping my head around existing and non-existing attributes and also we need to block "faulty" updates from the participant list update etc. But feel free to do it in a follow up PR when you think it's worth it

Reply:
Not sure if you understood what I meant.

It seems that both `isInCall` and `isConnected` always contain the exact same data "flags".
There is no scenario I could find where they would contain different data, so why use two attributes instead of just one ?

Reply:
the attribute could be called `callFlags[token]` since it contains the flags.

Reply:
ah! okay, found it: so `finishedConnecting` is then deleting the token entry after success. that's the bit I missed.

in that case maybe just store a boolean in `connecting` instead of all the flags to avoid confusion

Reply:
I'm fine leaving everything as is. it's just likely that the next dev coming over here might wonder as well